### PR TITLE
#draft rust api for picasso(s)/dali(s), fixed to `call` tool with examples in sudo.

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Description
* based on https://github.com/ComposableFi/composable/pull/1754 (please check description)
* basically tool to easy run TXes with copy paste and visual check + chain API generator project
* UNWRAPs are good enough errors for yet simple CLI tool

### CODE REVIEW

**Review by KaiserKarel — APPROVED** [review]

**Review by alexkdefi — APPROVED** [review]
👍

**Comment by dzmitry-lahoda** [comment]
@Mergifyio refresh

File: code/utils/xcmp/awesome-xcmp.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ XCMP can be upward (parachain to relay), downward(relay to parachain) and siblin
 - <https://www.youtube.com/watch?v=P_yLrFfmLrU>
 - <https://blog.quarkslab.com/resources/2022-02-27-xcmv2-audit/21-12-908-REP.pdf>
 - <https://github.com/paritytech/polkadot/blob/master/roadmap/implementers-guide/src/messaging.md>
+- https://forum.polkadot.network/t/how-xcm-will-actually-be-used-with-xcmp/190
```
Comment (andor0) [inline code/utils/xcmp/awesome-xcmp.md:42]:
```suggestion
- <https://forum.polkadot.network/t/how-xcm-will-actually-be-used-with-xcmp/190>
```

File: code/utils/xcmp/awesome-xcmp.md [hunk 2]
```diff
@@ -72,7 +73,7 @@ XCMP can be upward (parachain to relay), downward(relay to parachain) and siblin
 ### Operations
 
 <https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/>
-
+- https://substrate.stackexchange.com/questions/tagged/xcm
```
Comment (andor0) [inline code/utils/xcmp/awesome-xcmp.md:76]:
```suggestion
- <https://substrate.stackexchange.com/questions/tagged/xcm>
```

File: code/utils/xcmp/src/config.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	ReserveTransferNative(ReserveTransferNative),
+	Execute(Execute),
+}
+
+/// tries to parse and execute extrinsic again define chain
+#[derive(Parser, Debug)]
+pub struct Execute {
+	/// path to key
+	#[clap(long)]
+	pub suri: String, // name for parity clis
```
Comment (andor0) [inline code/utils/xcmp/src/config.rs:38]:
```suggestion
	pub suri: String, // name for parity CLIs
```

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 4]
```diff
@@ -61,210 +57,37 @@ The XCM message to Statemine is `0x1f00010100020c000400000000070010a5d4e81300000
...
+Which is:
+https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fpicasso-rpc.composable.finance%3A443#/extrinsics/decode/0x2302080602006d6f646c70792f747273727900000000000000000000000000000000000000000070617261e80300000000000000000000000000000000000000000000000000000b00b01723010a290001000100a10f0204060202286beef41f00010100020c000400000000070010a5d4e81300000000070010a5d4e800060002286bee5c1802083c01270800003c0027080000e803000000900100
+
+#### Decoded proposal
 
-Where [Kusama.utility.batchAll](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics) is encoded as `0x1802083c01270800003c0027080000e803000000900100`
+Above encoded call is has next XCM Transact message:
 
-## Accept request from Statemine
+https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fstatemine-rpc.dwellir.com#/extrinsics/decode/0x1f00010100020c000400000000070010a5d4e81300000000070010a5d4e800060002286bee5c1802083c01270800003c0027080000e803000000900100
```
Comment (andor0) [inline rfcs/0003-xcmp-picasso-statemine.md:71]:
```suggestion
<https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fstatemine-rpc.dwellir.com#/extrinsics/decode/0x1f00010100020c000400000000070010a5d4e81300000000070010a5d4e800060002286bee5c1802083c01270800003c0027080000e803000000900100>
```

File: rfcs/0003-xcmp-picasso-statemine.md [hunk 5]
```diff
@@ -61,210 +57,37 @@ The XCM message to Statemine is `0x1f00010100020c000400000000070010a5d4e81300000
...
-Accept channel from state mine as  `0x3c01e8030000` (can be decoded by Kusama)
+Where [Kusama.utility.batchAll](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics/decode/0x1802083c01270800003c0027080000e803000000900100) is encoded as ``
 
+## Picasso to accept request from Statemine
 
-Sent as XCM  message from Picasso with that acceptance via `0x2900010100020c0004000000000700e876481713000100000700e876481700060102286bee183c01e8030000`
+Accept channel from Statemine:
 
-Decoded by https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fpicasso-rpc.composable.finance#/extrinsics/decode
+https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics/decode/0x3c01e8030000
```
Comment (andor0) [inline rfcs/0003-xcmp-picasso-statemine.md:79]:
```suggestion
<https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.polkadot.io#/extrinsics/decode/0x3c01e8030000>
```

File: code/utils/xcmp/README.md [hunk 6]
```diff
@@ -1,7 +1,15 @@
 # Overview
 
-Opinionated XCM client to relay and composable which allows to handle local and remote transfers and inter network setups and overviews. Designed for maximal debug ability.
+Client tools.
+## Example
 
-Idea to be similar to `junod` in `cosmos` ecosystem.
+```shell
+cargo +nightly run sudo execute --suri PRIVATE_KEY_PATH --call 0x0008083432 --network composable_dali_on_parity_rococo --rpc wss://rpc.composablefinance.ninja:443
```
Comment (andor0) [inline code/utils/xcmp/README.md:7]:
```suggestion
cargo +nightly run xcmp sudo execute --suri PRIVATE_KEY_PATH --call 0x0008083432 --network composable_dali_on_parity_rococo --rpc wss://rpc.composablefinance.ninja:443
```

**Review by andor0 — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
Will update this after work is done to have subxt client as part of CI/CD

**Pull request closed without merging**